### PR TITLE
GH-2446 (main) Fixes warning for DELEGATES and also show bigger icon in mult…

### PR DESCRIPTION
**Pull request by biafra23**
Handles #2446

Changes proposed in this pull request:
- Fixes warning for DELEGATES and also show bigger icon in mult…
- Against `main` branch

### CODE REVIEW

File: Multisig/Data/Services/Utils/DelegateWarningCalculator.swift [hunk 1]
```diff
@@ -76,7 +76,7 @@ class DelegateWarningCalculator {
     ///   - addressInfoIndex: name index of addresses in the decoded transaction data
     /// - Returns: true if transaction is untrusted, false otherwise.
     static func isUntrusted(multiSendTx: MultiSendTx, addressInfoIndex: AddressInfoIndex?) -> Bool {
-        let result = multiSendTx.operation == .delegate && addressInfoIndex?.values[multiSendTx.to]?.addressInfo == nil
+        let result = multiSendTx.operation == .delegate
```
Comment (DmitryBespalov) [inline Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: this would make any delegate operation untrusted. I think this is not correct

Comment (DmitryBespalov) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: the problem is that if the transaction has the flag untrusted, it should still be untrusted. 

I think the fix should be in a different code line

Comment (DmitryBespalov) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: it seems like the line 23 is not triggered?

Comment (biafra23) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: This is only called on MultiSendTxs that have the DELEGATE operation. According to Ivan, they should be warned about.

Comment (DmitryBespalov) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: I have checked the updates, and indeed my understanding was outdated when I was writing the ticket description.

Your change is correct ✅

Comment (DmitryBespalov) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: However, that means we need to remove all checks of the name in the calculator file and in the tests 🤦

Comment (DmitryBespalov) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:78]: remove the `else` branch completely on line 33

**Comment by DmitryBespalov** [comment]
up-to-date requirements:
* warn when receiving the `trustedDelegateCallTarget`
* warn on any nested "delegate" operation in multi-sends

**Comment by biafra23** [comment]
I modified the code and added the TX from the bug report for unit tests.

File: Multisig/Data/Services/Utils/DelegateWarningCalculator.swift [hunk 2]
```diff
@@ -10,15 +10,17 @@ class DelegateWarningCalculator {
     private static let TRUSTED = false
 
     // MultiSendTx is untrusted: operation = delegate && 'to' address is not 'known'
-        // 'resolving the name': look up in the address info index
+    // 'resolving the name': look up in the address info index
     // Safe Tx is untrusted: operation = delegate && 'to' is not 'known'
-        // 'resolving name': 'to' object's .name is nil
+    // 'resolving name': 'to' object's .name is nil
```
Comment (DmitryBespalov) [inline Multisig/Data/Services/Utils/DelegateWarningCalculator.swift:14]: the comment should change, too, now it is not synced with the code.

MultiSendTx is untrusted iff there's any "delegate" operation in its transactions.

File: Multisig/Data/Services/Utils/DelegateWarningCalculator.swift [hunk 3]
```diff
@@ -57,9 +57,9 @@ class DelegateWarningCalculator {
                 // parameter is trusted when all sub-transactions are trusted.
                 return allTransactionsTrusted
 
-            // if it's nil (.none) or not decoded, then we say it's trusted
+                    // if it's nil (.none) or not decoded, then we say it's trusted
             case .none, .unknown:
-                return true
+                return UNTRUSTED
```
Comment (DmitryBespalov) [inline Multisig/Data/Services/Utils/DelegateWarningCalculator.swift]: hm, the comment says it's trusted, but the retunred value is "UNTRUSTED".

I think in this code block, it checks whether the tx is trusted, and not that it is untrusted.

**Review by DmitryBespalov — COMMENTED** [review]
sorry but some more changes needed

File: Multisig/Data/Services/Utils/DelegateWarningCalculator.swift (hunk 3 above)
Comment (biafra23) [reply Multisig/Data/Services/Utils/DelegateWarningCalculator.swift]: You're right. This should remain `true` and not be replaced by the names constant

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**